src/shogun/base/DynArray.h

```diff
@@ -21,6 +23,19 @@ namespace shogun
 {
 template <class T> class CDynamicArray;
 
+template <typename T> struct ptr_only
```

Comment:
This would need some docs.
And a SWIG ignore

also @lambday and @vigsterkr should comment